synapseclient/core/upload/upload_utils.py

```diff
@@ -2,9 +2,155 @@
 
 import math
 import re
+from io import BytesIO, StringIO
+from typing import Any, Dict, Optional, TypeVar, Union
+
+DATA_FRAME_TYPE = TypeVar("pd.DataFrame")
+
+
+def get_partial_dataframe_chunk(
+    df: DATA_FRAME_TYPE,
+    bytes_to_prepend: bytes,
+    part_number: int,
+    part_size: int,
+    byte_offset: int,
+    total_size_of_chunks_being_uploaded: int,
+    line_start: int,
+    line_end: int,
+    to_csv_kwargs: Optional[Dict[str, Any]] = None,
+) -> bytes:
+    """Read the nth chunk from the file assuming that we are not going to be reading
+    or uploading the entire file. This function allows us to read a portion of the file
+    and upload it to Synapse.
 
+    Arguments:
+        bytes_to_prepend: Bytes to prepend to the first chunk.
+        part_number: The part number.
+        part_size: The maximum size of the part to read for the upload process.
+        byte_offset: The byte offset for the file that has already been read and
+            and uploaded. This offset is used to calculate the total offset to read
+            the next chunk.
+        path_to_file_to_split: The path to the file that we are reading off disk and
+            uploading portions of to Synapse.
+        total_size_of_chunks_being_uploaded: The total size of the chunks that are being
+            uploaded. This is used to calculate the maximum number of bytes to read
+            from the file, accounting for the last chunk that may be smaller than the
+            chunk size.
+    """
+    header_bytes = None
+    if bytes_to_prepend and part_number == 1:
```

Comment:
This logical seems to imply that bytes_to_prepend can be `None`, but the type hints don't indicate it is optional. Does this check for an empty bytes object?

Reply:
Moved this to be Optional